databricks_cli/pipelines/api.py

```diff
@@ -41,17 +41,19 @@ def __init__(self, api_client):
         self.client = DeltaPipelinesService(api_client)
         self.dbfs_client = DbfsApi(api_client)
 
-    def deploy(self, spec, headers=None):
-        lib_objects = LibraryObject.from_json(spec.get('libraries', []))
-        local_lib_objects, external_lib_objects = \
-            self._identify_local_libraries(lib_objects)
+    def create(self, spec, headers=None):
+        spec = self._upload_libraries_and_update_spec(spec)
+        return self.client.client.perform_query('POST', '/pipelines', data=spec, headers=headers)
 
-        spec['libraries'] = LibraryObject.to_json(external_lib_objects +
-                                                  self._upload_local_libraries(local_lib_objects))
+    def deploy(self, spec, headers=None):
+        spec = self._upload_libraries_and_update_spec(spec)
         pipeline_id = spec['id']
         self.client.client.perform_query('PUT', '/pipelines/{}'.format(pipeline_id), data=spec,
                                          headers=headers)
 
+    def list(self, headers=None):
+        return self.client.client.perform_query("GET", "/pipelines", headers=headers)
```

Comment:
I thought we can't do this without pagination since it's a breaking API change? It feels wrong to put all this name validation logic in the CLI; it should be done on the deployment service. The CLI always tries to deploy a pipeline (optionally supplying the force or ignore_name_collision arg), and the deployment service decides whether or not to throw.